src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java

```diff
@@ -1448,4 +1471,169 @@ public void findDacByAccession() throws Exception {
                 .andExpect(jsonPath("$..dacs").isArray())
                 .andExpect(jsonPath("$..dacs.length()").value(0));
     }
+
+    @Test
+    public void postCrossReference() throws Exception {
+        String location = postTestCrossReference("DUO", "0000005");
+
+        mockMvc.perform(get(location))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id.db").value("DUO"))
+                .andExpect(jsonPath("$.id.id").value("0000005"));
+    }
+
+    private String postTestCrossReference(String db, String id) throws Exception {
+        DbIdVersionEntityId dbIdVersionEntityId = new DbIdVersionEntityId(db, id);
+        String jsonContent = "{ " +
+                "\"id\": " + testDbIdVersionEntityIdJson.write(dbIdVersionEntityId).getJson() + "," +
+                "\"label\": \"" + "label" + "\"," +
+                "\"url\": \"" + "url" + "\"" +
+                "}";
+
+        MvcResult mvcResult = mockMvc.perform(post("/crossReferences")
+                .content(jsonContent))
+                .andExpect(status().isCreated()).andReturn();
+
+        return mvcResult.getResponse().getHeader("Location");
+    }
+
+    @Test
+    public void postDuo() throws Exception {
+        String crUrl = postTestCrossReference("DUO", "0000007");
+        String duoUrl = postTestDuo(crUrl, null);
+
+        mockMvc.perform(get(duoUrl))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$..self.href").value(duoUrl));
+
+        mockMvc.perform(get(duoUrl + "/condition"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$..self.href").value(crUrl));
+
+        String modifierUrl = postTestCrossReference("DUO", "0000008");
+        String duoWithModifierUrl = postTestDuo(crUrl, Arrays.asList(modifierUrl));
```

Comment:
modification should be done by patch method, so if u kinda add auto id this should fail.

Reply:
No, this is not a modification to a created object. The `Duo` object happens to have a field called `modifiers` which is used optionally to further specifiy the field `condition`.  Here `Duo`s with same `condition`s but with different `modifiers` are treated as different objects.